### PR TITLE
Forward buffer

### PR DESCRIPTION
**Pull request by armstrmi**
## Description of Changes
* Entries are now flushed upon the closing of buffer for forward output
* Updated tests to verify forward output does send entries after closing
## **Please check that the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Add a changelog entry (for non-trivial bug fixes / features)
- [ ] CI passes

### CODE REVIEW

File: operator/builtin/output/forward/forward_test.go [hunk 1]
```diff
@@ -59,3 +59,31 @@ func TestForwardOutput(t *testing.T) {
...
+	newEntry := entry.New()
+	newEntry.Body = "test"
+	newEntry.Timestamp = newEntry.Timestamp.Round(time.Second)
+	err = forwardOutput.Process(forwardOutput.ctx, newEntry)
+	require.NoError(t, err)
+
+	err = forwardOutput.Stop()
+	require.NoError(t, err)
+
+	response := <-received
```
Comment (BinaryFissionGames) [inline operator/builtin/output/forward/forward_test.go:87]: Can we add a timeout on this receive? Like how it's done in the test above.

File: operator/builtin/output/forward/forward_test.go [hunk 2]
```diff
@@ -59,3 +59,31 @@ func TestForwardOutput(t *testing.T) {
...
+	ops, err := cfg.Build((testutil.NewBuildContext(t)))
+	require.NoError(t, err)
+
+	forwardOutput, ok := ops[0].(*ForwardOutput)
+	require.True(t, ok)
+
+	newEntry := entry.New()
+	newEntry.Body = "test"
+	newEntry.Timestamp = newEntry.Timestamp.Round(time.Second)
+	err = forwardOutput.Process(forwardOutput.ctx, newEntry)
```
Comment (BinaryFissionGames) [inline operator/builtin/output/forward/forward_test.go:81]: I think the context here should actually be `context.Background()`, not using the internal context.

**Review by BinaryFissionGames — COMMENTED** [review]
Just have a few small nitpicks with the test, then it should be good to merge.

**Review by BinaryFissionGames — APPROVED** [review]
LGTM!

**Review by cpheps — CHANGES_REQUESTED** [review]

File: operator/builtin/output/forward/forward_test.go [hunk 3]
```diff
@@ -59,3 +59,31 @@ func TestForwardOutput(t *testing.T) {
 		require.Equal(t, newEntry.Resource, e.Resource)
 	}
 }
+
+func TestFlushBufferOnClose(t *testing.T) {
+	cfg := NewForwardOutputConfig("test")
+	received := make(chan []byte, 1)
+	srv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
```
[the remark this answers is not in the document]
Comment (cpheps) [reply operator/builtin/output/forward/forward_test.go:66]: I'll fix this as I need to rebase on some fixes I made that were breaking tests.

**Review by cpheps — APPROVED** [review]
LGTM

**Comment by cpheps** [comment]
Test failures are unrelated to code changes.

**Pull request merged**